src/modules/avformat/consumer_avformat.c

```diff
@@ -1131,14 +1104,23 @@ typedef struct encode_ctx_desc
 static int encode_audio(encode_ctx_t* ctx)
 {
 	char key[27];
-	int i, j = 0, samples, n;
+	int i, j = 0, samples;
+
+	int frame_length = ctx->audio_input_frame_size * ctx->channels * ctx->sample_bytes;
 
-	n = FFMIN( FFMIN( ctx->channels * ctx->audio_input_nb_samples, sample_fifo_used( ctx->fifo ) / ctx->sample_bytes ), AUDIO_ENCODE_BUFFER_SIZE );
+	// Get samples count to fetch from fifo
+	if ( sample_fifo_used( ctx->fifo ) < frame_length )
+		samples = sample_fifo_used( ctx->fifo ) / ( ctx->channels * ctx->sample_bytes );
+	else
+	{
+		int frames = FFMIN( sample_fifo_used( ctx->fifo ), AUDIO_ENCODE_BUFFER_SIZE ) / frame_length;
+		samples = frames * ctx->audio_input_frame_size;
```

Comment:
Many codecs cannot accept more than one frame, but this multiplies the AVCodec.frame_size by `frames`, which goes into into AVFrame.nb_samples below.

Reply:
Removing `frames *` ends up feeding PCM one sample (*channels) at-a-time, which is not efficient.